addons/stock/static/src/js/report_stock_forecasted.js

```diff
@@ -137,22 +137,34 @@ const ReplenishReport = clientAction.extend({
     },
 
     /**
-     * Returns a domain to filter on the product variant or product template
+     * Returns a Promise with the given viewInfo and
+     * the domain to filter on the product variant or product template
      * depending of the active model.
      *
-     * @returns {Array}
+     * Injecting the product_ids when resModel === 'product.template'
+     * improves report.stock.quantity read_group perfs.
+     * 
+     * @returns {Promise}
      */
     _getReportDomain: function () {
         const domain = [
             ['state', '=', 'forecast'],
             ['warehouse_id', '=', this.active_warehouse.id],
         ];
         if (this.resModel === 'product.template') {
-            domain.push(['product_tmpl_id', '=', this.productId]);
+            return this._rpc({
+                model: 'product.product',
+                method: 'search',
+                args: [[['product_tmpl_id', '=', this.productId]]],
+                context: {active_test: false},
+            }).then((ids) => {
+                domain.push(['product_id', 'in', ids])
```

Comment:
This may not work well if product has 1000 variants. Adding related field to the view looks as a valid idea.